File: mealwurm-client/src/types/Ingredient.ts
export const Units = {
  0: "NONE",
  1: "CUP",
  2: "GALLON",
  3: "GRAM",
  4: "KILOGRAM",
  5: "LITER",
  6: "MILLIGRAM",
  7: "MILLILITER",
  8: "OUNCE",
  9: "PINT",
  10: "POUND",
  11: "QUART",
  12: "TABLESPOON",
  13: "TEASPOON",
};

export type Ingredient = {
  id: string;
  name: string;
  quantity: number;
  unit: number; // maps to UNITS
  modifier: string;
  createdAt: string; // ISO 8601 format
  createdBy: string; // UUID
  lastModifiedAt: string; // ISO 8601 format
  lastModifiedBy: string; // UUID
};

File: mealwurm-server/src/main/java/com/mealwurm/server/model/Ingredient.java
package com.mealwurm.server.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@EqualsAndHashCode(callSuper = true)
@Data
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Ingredient extends BaseEntity {
    public enum Unit {
        NONE,
        CUP,
        GALLON,
        GRAM,
        KILOGRAM,
        LITER,
        MILLIGRAM,
        MILLILITER,
        OUNCE,
        PINT,
        POUND,
        QUART,
        TABLESPOON,
        TEASPOON
    }

    private String name;
    // divided, chilled, etc.
    private String modifier;
    private Unit unit;
    private float quantity;
    private int ingredientOrder;

    // many ingredients to one recipe
    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "recipe_id")
    private Recipe recipe;
}
